server/app/models/Application.java

```diff
@@ -93,6 +93,10 @@ public Program getProgram() {
     return this.program;
   }
 
+  public String getProgramName() {
```

Comment:
**suggestion (blocking):** add javadoc for this new public method. All public methods should have javadoc, though some are still missing in the codebase.